src/rabbit.erl

```diff
@@ -442,15 +448,53 @@ environment(App) ->
     lists:keysort(1, [P || P = {K, _} <- application:get_all_env(App),
                            not lists:member(K, Ignore)]).
 
+
 rotate_logs(BinarySuffix) ->
+    rabbit_log:info("Rotating logs with suffix '~s'~n", [BinarySuffix]),
     Suffix = binary_to_list(BinarySuffix),
-    rabbit_log:info("Rotating logs with suffix '~s'~n", [Suffix]),
-    log_rotation_result(rotate_logs(log_location(kernel),
-                                    Suffix,
-                                    rabbit_error_logger_file_h),
-                        rotate_logs(log_location(sasl),
-                                    Suffix,
-                                    rabbit_sasl_report_file_h)).
+    ok = rotate_lager_handlers(Suffix, application:get_env(lager, handlers)),
+    ok = rotate_lager_sinks(Suffix),
+    %% give Lager a moment to perform rotation
+    timer:sleep(2000),
```

Comment:
-- Comment from outdated diff --
Sleep wait for lager to switch to new files. Nasty solution, but log rotation are called only from rabbitmqctl (should be deprecated maybe?) 
Function itself required for backwards compatibility because used in logrotate scripts in Linux packages. 
Lager internal log rotation is much nicer because support file handle management and multiple rotated files, size, date and other settings. Maybe we should recommend to use it instead?